cf-custom-resources/lib/custom-domain.js

```diff
@@ -0,0 +1,328 @@
+      case aliasTypes.RootDomainZone:
+        await writeARecord(
+          appRoute53,
+          alias,
+          lbDNS,
+          lbHostedZone,
+          aliasType.domain,
+          action
+        );
+        break;
+      // We'll skip if it is the other alias type since it will be in another account's route53.
+      default:
+    }
+  }
+};
+
+const writeARecord = async function (
+  route53,
+  alias,
+  lbDNS,
+  lbHostedZone,
+  domain,
+  action
+) {
+  const hostedZones = await route53
+    .listHostedZonesByName({
+      DNSName: domain,
+      MaxItems: "1",
+    })
+    .promise();
+
+  if (!hostedZones.HostedZones || hostedZones.HostedZones.length == 0) {
+    throw new Error(`Couldn't find any Hosted Zone with DNS name ${domain}.`);
+  }
+  const hostedZoneId = hostedZones.HostedZones[0].Id.split("/").pop();
+  console.log(`${action} A record into Hosted Zone ${hostedZoneId}`);
+  const changeBatch = await updateRecords(
+    route53,
+    hostedZoneId,
+    action,
+    alias,
+    lbDNS,
+    lbHostedZone
+  );
+  await waitForRecordChange(route53, changeBatch.ChangeInfo.Id);
+};
+
+/**
+ * Custom domain handler, invoked by Lambda.
+ */
+exports.handler = async function (event, context) {
+  var responseData = {};
+  var physicalResourceId;
+  const props = event.ResourceProperties;
+  const [app, env, domain] = [props.AppName, props.EnvName, props.DomainName];
+  var aliasTypes = {
+    EnvDomainZone: {
+      regex: `.*${env}.${app}.${domain}`,
+      domain: `${env}.${app}.${domain}`,
+    },
+    AppDomainZone: { regex: `.*${app}.${domain}`, domain: `${app}.${domain}` },
+    RootDomainZone: { regex: `.*${domain}`, domain: `${domain}` },
+    OtherDomainZone: { regex: `.*` },
+  };
+
+  try {
+    switch (event.RequestType) {
+      case "Create":
+      case "Update":
+        await writeCustomDomainRecord(
+          props.Aliases,
+          props.LoadBalancerDNS,
+          props.LoadBalancerHostedZone,
+          props.AppDNSRole,
+          aliasTypes,
+          "UPSERT"
+        );
+        physicalResourceId = `custom-domain-${event.LogicalResourceId}`;
+        break;
+      case "Delete":
+        await writeCustomDomainRecord(
+          props.Aliases,
+          props.LoadBalancerDNS,
+          props.LoadBalancerHostedZone,
+          props.AppDNSRole,
+          aliasTypes,
+          "DELETE"
+        );
+        physicalResourceId = event.PhysicalResourceId;
+        break;
+      default:
+        throw new Error(`Unsupported request type ${event.RequestType}`);
+    }
+    await report(event, context, "SUCCESS", physicalResourceId, responseData);
+  } catch (err) {
+    console.log(`Caught error ${err}.`);
+    await report(
+      event,
+      context,
+      "FAILED",
+      physicalResourceId,
+      null,
+      err.message
+    );
+  }
+};
+
+// getAllAliases gets all aliases out from a string. For example:
+// {"frontend": "foobar.com,barfoo.com", "api": "foo.com"} will return
```

Comment:
I'm trying to refresh my memory, is the idea that this would be more realistically:
```suggestion
// {"frontend": "test.foobar.com,foobar.com", "api": "api.foobar.com"} will return
```
i.e. a service will have multiple aliases because of environments overrides